File: algorithms/leetcode/lc008_string_to_integer_a_to_i/LC008StringToIntegerAToI.Tests.cs
using System;
using Xunit;

namespace AlgorithmDesign.leetcode.lc008_string_to_integer_a_to_i
{
    public class SolutionTests
    {
        [Fact]
        public void TestMinusMinus3241()
        {
            Assert.Equal(0, new Solution().MyAtoi("--3241"));
        }

        [Fact]
        public void TestPlusPlus3241()
        {
            Assert.Equal(0, new Solution().MyAtoi("++3241"));
        }

        [Fact]
        public void TestMinusPlus3241()
        {
            Assert.Equal(0, new Solution().MyAtoi("-+3241"));
        }

        [Fact]
        public void TestMinus3241()
        {
            Assert.Equal(-3241, new Solution().MyAtoi("-3241"));
        }

        [Fact]
        public void TestSpaceMinusPlus3241A()
        {
            Assert.Equal(-3241, new Solution().MyAtoi(" -3241a"));
        }

        [Fact]
        public void Test9223372036854775809()
        {
            Assert.Equal(Int32.MaxValue, new Solution().MyAtoi("9223372036854775809"));
        }

        [Fact]
        public void TestMinus9223372036854775809()
        {
            Assert.Equal(Int32.MinValue, new Solution().MyAtoi("-9223372036854775809"));
        }

        [Fact]
        public void TestNothing()
        {
            Assert.Equal(0, new Solution().MyAtoi("nothing"));
        }
    }
}

File: algorithms/code/leetcode/lc008_string_to_integer_a_to_i/LC008StringToIntegerAToITests.java
package leetcode.lc008_string_to_integer_a_to_i;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public final class LC008StringToIntegerAToITests {
    @Test
    public void testMinusMinus3241() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(0, solution.myAtoi("--3241"));
    }

    @Test
    public void testPlusPlus3241() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(0, solution.myAtoi("++3241"));
    }

    @Test
    public void testMinusPlus3241() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(0, solution.myAtoi("-+3241"));
    }

    @Test
    public void testMinus3241() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(-3241, solution.myAtoi("-3241"));
    }

    @Test
    public void testSpaceMinusPlus3241A() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(-3241, solution.myAtoi(" -3241a"));
    }

    @Test
    public void test9223372036854775809() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(Integer.MAX_VALUE, solution.myAtoi("9223372036854775809"));
    }

    @Test
    public void testMinus9223372036854775809() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(Integer.MIN_VALUE, solution.myAtoi("-9223372036854775809"));
    }

    @Test
    public void testNothing() throws Exception {
        LC008StringToIntegerAToI solution;
        solution = new LC008StringToIntegerAToI();
        assertEquals(0, solution.myAtoi("nothing"));
    }
}
